### PR TITLE
SHOW TYPES and \dT

### LINKED ISSUE #8398: sql: SHOW TYPES & cli: add the postgres \dT alias for listing enums
PR: https://github.com/cockroachdb/cockroach/pull/53386

From release notes:
> CockroachDB now recognizes the `\dT` alias for listing user-defined [types](../v20.2/data-types.html). [#53386][#53386] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #8398.

- Added `SHOW TYPES` page.
- Added `\dT` command to `cockroach sql` page.
- Made some corrections/clarifications on a couple pages.

### CODE REVIEW

File: v20.2/cockroach-sql.md [hunk 1]
```diff
@@ -146,6 +146,7 @@ Command | Usage
 `\h <statement>`<br>`\hf <function>` | View help for specific SQL statements or functions. See [SQL shell help](#help) for more details.
 `\l` | List all databases in the CockroachDB cluster. This command is equivalent to [`SHOW DATABASES`](show-databases.html).
 `\dt`<br>`d` | Show the tables of the current schema in the current database. These commands are equivalent to [`SHOW TABLES`](show-tables.html).
+`\dT` | <span class="version-tag">New in v20.2:</span> Show the [user-defined types](enum.html) in the current database. This command is equivalent to [`SHOW ENUMS`](show-enums.html) and [`SHOW TYPES`](show-types.html).
```
Comment (knz) [inline v20.2/cockroach-sql.md:149]: you should keep that to just SHOW TYPES to not give the mistaken impression that \dT is specific to enums

**Review by knz — APPROVED** [review]
LGTM with nit

File: v20.2/cockroach-sql.md (hunk 1 above)
Comment (ericharmeling) [reply v20.2/cockroach-sql.md]: Done. TFTR!

**Review by lnhsingh — COMMENTED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8496#-:-MIUn7IHC96PrmvZ7AcO:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @knz and @lnhsingh)

**Pull request merged**